gateway.go

```diff
@@ -248,7 +264,13 @@ routes[{{ method . }}] = append(routes[{{ method . }}], &gatewayRoute{regexp.Mus
 					query := r.URL.Query()
 					for path := range query {
 						// TODO(shane): Freak out on error!
-						gatewaySetJSON(request, strings.Trim(gr.body + "." + path, "."), query.Get(path))
+						var value interface{} = query.Get(path)
+						switch gr.inputFieldTypes[path] {
+						case descriptor.FieldDescriptorProto_TYPE_BOOL:
+							value = value == "true" || value == "1"
```

Comment:
Will the value received always be a string? What would happen if a bool is passed as 1 without the quotes.
```
"isBool": 1
```
Would this still evaluate to true for `value == "1"`?